### PR TITLE
Test/implement tests

### PR DESCRIPTION
**Pull request by iannottivictoria**
Added the tests (unit and integration).

### CODE REVIEW

File: src/test/java/com/ecse437/project/ProjectApplicationTests.java [hunk 1]
```diff
@@ -1,13 +1,85 @@
...
+	
 	@Test
 	void contextLoads() {
+		assertThat(controller).isNotNull();
+	}
+	
+	@Test
+	void binaryCalculation() {
+		binary = new Binary();
+		assertThat(binary.compute(255).toString()).isEqualTo("11111111");
```
Comment (ryadammar1) [inline src/test/java/com/ecse437/project/ProjectApplicationTests.java:49]: no need for toString()

File: src/test/java/com/ecse437/project/ProjectApplicationTests.java [hunk 2]
```diff
@@ -1,13 +1,85 @@
...
+import com.ecse437.project.services.Squared;
+
 @SpringBootTest
 class ProjectApplicationTests {
 
+	@Autowired
+	private ProjectController controller;
+	
+	@Mock
+	private Binary binary;
```
Comment (ryadammar1) [inline src/test/java/com/ecse437/project/ProjectApplicationTests.java:27]: we are mocking these services but never using the mocks

File: src/test/java/com/ecse437/project/ProjectApplicationTests.java [hunk 3]
```diff
@@ -1,13 +1,85 @@
...
+	
+	@Mock
+	private Prime prime;
+	
+	@Mock
+	private Squared squared;
+	
 	@Test
 	void contextLoads() {
+		assertThat(controller).isNotNull();
```
Comment (ryadammar1) [inline src/test/java/com/ecse437/project/ProjectApplicationTests.java:43]: This test doesn't check for anything useful. If you want to test the controller, test the endpoints

File: src/test/java/com/ecse437/project/ProjectApplicationTests.java [hunk 4]
```diff
@@ -1,13 +1,85 @@
...
+import com.ecse437.project.exceptions.IllegalAlgorithmException;
+import com.ecse437.project.services.Binary;
+import com.ecse437.project.services.Factorial;
+import com.ecse437.project.services.Hexa;
+import com.ecse437.project.services.Main;
+import com.ecse437.project.services.Prime;
+import com.ecse437.project.services.Squared;
+
 @SpringBootTest
 class ProjectApplicationTests {
```
Comment (ryadammar1) [inline src/test/java/com/ecse437/project/ProjectApplicationTests.java:21]: All these tests should be refactored into different classes. We especially want to separate integration and unit tests

File: src/test/java/com/ecse437/project/ProjectApplicationTests.java [hunk 5]
```diff
@@ -1,13 +1,85 @@
...
+	
+	@Test
+	void SquaredCalculation() {
+		squared = new Squared();
+		assertThat(squared.compute(122).toString()).isEqualTo("14884");
+	}
+	 
+	@Test
+	void SystemIntegration() throws IllegalAlgorithmException {
+		assertThat(Main.generateAlgorithmOutput(Algorithm.Factorial).toString()).isNotNull();
```
Comment (ryadammar1) [inline src/test/java/com/ecse437/project/ProjectApplicationTests.java:78]: Let's test for `IllegalAlgorithmException`

File: src/test/java/com/ecse437/project/ProjectApplicationTests.java [hunk 6]
```diff
@@ -1,13 +1,85 @@
...
+		assertThat(Main.generateAlgorithmOutput(Algorithm.Prime).toString()).isNotNull();
```
Comment (ryadammar1) [inline src/test/java/com/ecse437/project/ProjectApplicationTests.java:79]: Instead of simply checking for isNotNull, we can mock the Random class to provided specific outputs and assert for them. This would require that we modify the main class a bit to use a custom Random class wrapper.

e.g.

Add

```
class MyRandomUtility {

  public Int getRandom(){
      java.util.Random.nextInt().....
  }
}
```

to the Common class and instantiate the class within main before calling getRandom()

File: src/test/java/com/ecse437/project/ProjectApplicationTests.java (hunk 1 above)
Comment (ryadammar1) [inline src/test/java/com/ecse437/project/ProjectApplicationTests.java:49]: Worth have more cases for the `compute` function. I.e. test boundary values, edge cases etc

Comment (ryadammar1) [inline src/test/java/com/ecse437/project/ProjectApplicationTests.java:49]: Also worth testing the `print` function to test that the formatting is right

**Review by ryadammar1 — CHANGES_REQUESTED** [review]

File: src/test/java/com/ecse437/project/ProjectApplicationTests.java (hunk 1 above)
Comment (iannottivictoria) [reply src/test/java/com/ecse437/project/ProjectApplicationTests.java:49]: Removed "toString()"

File: src/test/java/com/ecse437/project/ProjectApplicationTests.java (hunk 5 above)
Comment (iannottivictoria) [reply src/test/java/com/ecse437/project/ProjectApplicationTests.java:78]: Added a test for IllegalAlgorithmException

File: src/test/java/com/ecse437/project/ProjectApplicationTests.java (hunk 1 above)
Comment (iannottivictoria) [reply src/test/java/com/ecse437/project/ProjectApplicationTests.java:49]: Added more test cases for the compute function for the different functions

Comment (iannottivictoria) [reply src/test/java/com/ecse437/project/ProjectApplicationTests.java:49]: Tested the print function for each function

File: src/test/java/com/ecse437/project/ProjectApplicationTests.java (hunk 6 above)
Comment (iannottivictoria) [reply src/test/java/com/ecse437/project/ProjectApplicationTests.java:79]: Will complete this refactor separately.

**Comment by iannottivictoria** [comment]
The bugfix/functions_fix PR needs to go through first.

File: src/test/java/com/ecse437/project/IllegalAlgorithmExceptionTests.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+		  
+	@Test
+	void ExceptionThrown() {
+
+		Exception exception = assertThrows(IllegalAlgorithmException.class, () -> 
+		{
+			throw new IllegalAlgorithmException(Algorithm.Binary);
+		});
+		
+		String expectedMessage = "Unsupported algorithm: ";
```
Comment (ryadammar1) [inline src/test/java/com/ecse437/project/IllegalAlgorithmExceptionTests.java:23]: Nice test, it exposed a bug - It should be "Unsupported algorithm: Binary".

**Review by ryadammar1 — COMMENTED** [review]
I'm guessing this is only unit testing so far? Needs integration testing

**Review by ryadammar1 — APPROVED** [review]
LGTM
